internal/rename/application.go

```diff
@@ -0,0 +1,69 @@
+/*
+ *  *******************************************************************************
+ *  * Copyright (c) 2019 Edgeworx, Inc.
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Eclipse Public License v. 2.0 which is available at
+ *  * http://www.eclipse.org/legal/epl-2.0
+ *  *
+ *  * SPDX-License-Identifier: EPL-2.0
+ *  *******************************************************************************
+ *
+ */
+
+package rename
+
+import (
+	"fmt"
+	apps "github.com/eclipse-iofog/iofog-go-sdk/pkg/apps"
+	"github.com/eclipse-iofog/iofog-go-sdk/pkg/client"
+	"github.com/eclipse-iofog/iofogctl/internal"
+	"github.com/eclipse-iofog/iofogctl/internal/config"
+	"github.com/eclipse-iofog/iofogctl/pkg/util"
+)
+
+type applicationExecutor struct {
+	namespace string
+	name      string
+	newname   string
+}
+
+func newApplicationExecutor(namespace, name, newname string) *applicationExecutor {
+	a := &applicationExecutor{}
+	a.namespace = namespace
+	a.name = name
+	a.newname = newname
+	return a
+}
+
+func (exe *applicationExecutor) GetName() string {
+	return exe.name
+}
+
+func (exe *applicationExecutor) Execute() error {
+	// Get config
+	agent, err := config.GetAgent(exe.namespace, exe.name)
+	if err != nil {
+		return err
+	}
+
+	util.SpinStart(fmt.Sprintf("Renaming Agent %s", name))
+
+	// Init remote resources
+	clt, err := internal.NewControllerClient(exe.namespace)
+	if err != nil {
+		return err
+	}
+
+	err = RenameAgent(exe.namespace, exe.name, exe.newname)
+	if err != nil {
+		util.printError(err)
+	}
+
+	_, err = clt.UpdateAgent(client.AgentUpdateRequest{
```

Comment:
Same, why does application call agent update ?

Reply:
UpdateAgent is to take the same changes we've made shallowly, and update how the Controller sees the agent, if this is incorrect, I'm glad to change this

Reply:
Why are we trying to update an agent when we rename an application ?